### PR TITLE
Generate ids for dropdown component

### LINKED ISSUE #469: Add random ID to dropdown component
From https://github.com/exercism/website/pull/405.

### PR DESCRIPTION
**Pull request by kntsoriano**
Closes https://github.com/exercism/website/issues/469.

### CODE REVIEW

File: app/javascript/components/dropdowns/Dropdown.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
-import React, { useState, useEffect, useRef } from 'react'
+import React, { useState, useEffect, useRef, useMemo } from 'react'
 import { usePanel } from '../../hooks/use-panel'
+import { v4 as uuidv4 } from 'uuid'
```
Comment (ErikSchierboom) [inline app/javascript/components/dropdowns/Dropdown.tsx:3]: Good idea to rename!

File: test/javascript/components/dropdowns/Dropdown.test.js [hunk 2]
```diff
@@ -7,7 +7,6 @@ import userEvent from '@testing-library/user-event'
 test('down arrow opens menu on first item', async () => {
```
Comment (ErikSchierboom) [inline test/javascript/components/dropdowns/Dropdown.test.js:7]: Would it make sense to add a test to verify that the ID is generated randomkly?

**Review by ErikSchierboom — APPROVED** [review]

**Review by iHiD — APPROVED** [review]
This is great. Thanks :)

File: app/javascript/components/dropdowns/Dropdown.tsx (hunk 1 above)
Comment (SleeplessByte) [reply app/javascript/components/dropdowns/Dropdown.tsx:3]: image

From the docs 🖌️

File: test/javascript/components/dropdowns/Dropdown.test.js (hunk 2 above)
Comment (SleeplessByte) [reply test/javascript/components/dropdowns/Dropdown.test.js:7]: Meh, it's more important that each item has a unique ID, than that they are random. I would test: each item has an ID and non of the ID is the same.

**Pull request merged**